gateway/reverse_proxy.go

```diff
@@ -223,27 +223,38 @@ func (gw *Gateway) TykNewSingleHostReverseProxy(target *url.URL, spec *APISpec,
 		}()
 	})
 
+	if logger == nil {
+		logger = logrus.NewEntry(log)
+	}
+
+	logger = logger.WithField("mw", "ReverseProxy")
+
 	targetQuery := target.RawQuery
 	director := func(req *http.Request) {
+		logger := logger
+		spec := spec
+		target := target
+		gw := gw
```

Comment:
do we need to scope `gw` and `spec`? 
both are pointers.

Reply:
technically yes, because we don't know what happens outside, spec may be modified, gw may be modified and it would produce a race condition (on shutdown/reload), and the target produces one for sure; logger may not be needed since it's closure-local, but it also doesn't hurt.